### PR TITLE
Register new account/Login

### PR DESCRIPTION
**Pull request by VivianLi98**
Added registration and login functionality. Errors will be displayed via Snackbar. 

I had to changed some of repository methods to suspend functions since database can't be modified on main thread.

https://user-images.githubusercontent.com/103967461/166556041-1b783848-992d-41c9-a4a6-45733e294e01.mov

### CODE REVIEW

File: app/src/main/java/com/example/voyagerx/RegisterFragment.kt [hunk 1]
```diff
@@ -1,24 +1,86 @@
...
+                        it.findViewById(R.id.loginActivity),
+                        e.message ?: "Error",
+                        Snackbar.LENGTH_SHORT
+                    ).show()
+                }
+            }
+        }
+    }
+
+    private fun validate(email: String, password: String, confirmPassword: String){
```
Comment (brownyte) [inline app/src/main/java/com/example/voyagerx/RegisterFragment.kt:72]: I did not know about Patterns. Glad to have seen you expose this to me

File: app/src/main/java/com/example/voyagerx/repository/UserRepository.kt [hunk 2]
```diff
@@ -70,13 +81,15 @@ class UserRepository @Inject constructor(
...
+            }
 
-        database.userDao().insertNoProfile(email, password)
-        logIn(email, password)
+            database.userDao().insertNoProfile(email, password)
+            logIn(email, password)
+        }
     }
 
     fun getCurrentUser(): User?{
```
Comment (brownyte) [inline app/src/main/java/com/example/voyagerx/repository/UserRepository.kt:95]: @VivianLi98, Can I use this function to grab the user details for the profile screen? Sorry to ask probably a dumb question but I am walking through your code and getting acquainted with the database.

**Review by brownyte — APPROVED** [review]
@VivianLi98, very informative code! 

Can you tell me what "val listType = object: TypeToken<List<String>>(){}.type" is doing in your Converters.kt file (line 18)? I know that isn't in this commit but I am looking at the file now and confused about that code does.

File: app/src/main/java/com/example/voyagerx/repository/UserRepository.kt (hunk 2 above)
Comment (VivianLi98) [reply app/src/main/java/com/example/voyagerx/repository/UserRepository.kt:97]: Not a stupid question at all! There is getCurrUser() function in the UserRepository so you can just get it from there. You do need to inject the UserRepository though, but it's pretty straight forward. Just add a @AndroidEntryPoint and a @Inject annotation. You can take a look at my code to see how I did it.

Comment (brownyte) [reply app/src/main/java/com/example/voyagerx/repository/UserRepository.kt:97]: Thank you, @VivianLi98!!

**Pull request merged**